python/hsfs/engine/spark.py

```diff
@@ -490,13 +491,17 @@ def _random_split(self, dataset, training_dataset):
     def _time_series_split(
         self, training_dataset, dataset, event_time, drop_event_time=False
     ):
+        # registering the UDF
+        _convert_event_time_to_timestamp = udf(
+            util.convert_event_time_to_timestamp, LongType()
+        )
+
         result_dfs = {}
         ts_type = dataset.select(event_time).dtypes[0][1]
         ts_col = (
             unix_timestamp(col(event_time)) * 1000
```

Comment:
the udf should be able to handle `"date", "timestamp"` as well, so we do not need the if-clause.